[1.9.20-1.0.14] Error: Rewrite at slice LEXICAL_SCOPE key: ANNOTATION_ENTRY old value

```
error: java.lang.AssertionError: Rewrite at slice LEXICAL_SCOPE key: ANNOTATION_ENTRY old value:
org.jetbrains.kotlin.resolve.scopes.LexicalScope$Base@6f93355@116994901
new value: org.jetbrains.kotlin.resolve.scopes.LexicalScope$Base@3568912a@896045354
```

Seems to only happen on iOS (maybe all Kotlin Native). 

Repro: https://github.com/chrisbanes/tivi/pull/1620

--- a/compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/binary/KSAnnotationDescriptorImpl.kt
+++ b/compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/binary/KSAnnotationDescriptorImpl.kt
@@ -193,9 +193,10 @@ fun LazyAnnotationDescriptor.getValueArguments(): Map<Name, ConstantValue<*>> {
         override fun toString(): String = "${name.asString()} declared in LazyAnnotations.kt"
     }
 
-    val scope = (c.scope.ownerDescriptor as? PackageFragmentDescriptor)?.let {
-        LexicalScope.Base(c.scope, FileDescriptorForVisibilityChecks(source, it))
-    } ?: c.scope
+    val scope = c.trace.get(BindingContext.LEXICAL_SCOPE, annotationEntry)
+        ?: (c.scope.ownerDescriptor as? PackageFragmentDescriptor)?.let {
+            LexicalScope.Base(c.scope, FileDescriptorForVisibilityChecks(source, it))
+        } ?: c.scope
 
     val resolutionResults = c.annotationResolver.resolveAnnotationCall(annotationEntry, scope, c.trace)
     AnnotationResolverImpl.checkAnnotationType(annotationEntry, c.trace, resolutionResults)

